fix: ensure we are placing the F2 in the correct position

claire-labry opened the pull request:
Ensure we're placing the F2 in the correct position.

claire-labry committed:
fix: ensure we are placing the F2 in the correct position

claire-labry committed:
add changelog notes

claire-labry committed:
add changelog notes

shore commented on .changes/unreleased/Changed-20251030-104246.yaml:
```diff
@@ -0,0 +1,3 @@
+kind: Changed
```
Because we're trying to _maintain_ a specific behavior, this should be `Fixed` (thus corresponding to a patch version bump rather than a major version bump).

claire-labry committed:
add changelog notes

Commit:
Support testing all Linux packages for variant tag placement

* Add helper to extract version string (identify its start, really)
* Add assertion to check that variant tags like F2 don't appear in the
  version string
* Call the new assertion only for Linux packages, because the variant
  tag is not moved before the version string for other artifact types
* Apply this for _all_ Linux package test cases

claire-labry committed:
remove test in favor of more accurate test

claire-labry committed:
new changelog note

claire-labry committed:
s'moar fixes

claire-labry committed:
appease the linter

claire-labry commented:
@shore I think this is ready for your review 🙏🏻

ethel-hashicorp commented on tests/translate-artifact-name.bats:
```diff
@@ -24,6 +52,20 @@ assert_string_absent() {
 
+# Check that variant tags like H and F2 are placed after the product name rather than after the version string.
+# This only applies to Linux packages.
+assert_linux_package_variant_placement() {
```
I really appreciate this clearly named helper function 👍

ethel-hashicorp commented on rename-by-part-number/action.yml:
```diff
@@ -13,15 +13,12 @@ inputs:
   pao_meta_dir:
     description: 'The path to the PAO meta directory'
     required: true
-    type: string
```
Non-blocking question: I'm just curious - what is your reasoning is for removing the explicit types here?

shore replied:
They are invalid.  The schema does not allow a `type` field here.
